File: solutions/c/leap/1/leap.c
#include "leap.h"

bool leap_year(int year) {
    bool is_leap = 0;
    if ( year % 100 == 0 ) {
        if ( year % 400 == 0 ) {
            is_leap = 1;
        }
    } else if ( year % 4 == 0 ) {
        is_leap = 1;
    }
    return is_leap;
}

File: solutions/erlang/leap/1/src/leap.erl
-module(leap).

-export([leap_year/1]).

leap_year(_Year) ->
    if ( _Year rem 4 =:= 0
         andalso ( _Year rem 100 =/= 0
                   orelse ( _Year rem 100 =:= 0
                            andalso _Year rem 400 =:= 0 ) )
       ) ->
            true;
       true ->
            false
    end
.
